### PR TITLE
doc test: Generalize doc test

### PR DESCRIPTION
**Pull request by TimePrinciple**
### Summary of the PR

- RISC-V does not implement `KVM_CAP_MAX_VCPU_ID`, use `MaxVcpus`
capability to test `check_extension` functionality on all platforms.
- According to warning emitted by cargo, replacing `config` with
`config.toml` since it's deprecated.
- Removed unnecessary target arhictecture predicates since the related
methods would not present for disabled architecture. Disabled 
`KVM_CAP_ENABLE_CAP` and g/set method of `kvm_regs` for RISC-V platform.
- As this patch [1] documents, and pointed out by @roypat, create an
in-kernel `irqchip` before invoking any ioctls relate to PIT.

[1] https://lore.kernel.org/kvm/CA+i-1C1LOn19FcddyC5kV8idGQq5KDdAjWBo80ANpRGn8DCx3g@mail.gmail.com/T/

### Requirements

Before submitting your PR, please make sure you addressed the following
requirements:

- [x] All commits in this PR have Signed-Off-By trailers (with
  `git commit -s`), and the commit message has max 60 characters for the
  summary and max 75 characters for each description line.
- [x] All added/changed functionality has a corresponding unit/integration
  test.
- [x] All added/changed public-facing functionality has entries in the "Upcoming 
  Release" section of CHANGELOG.md (if no such section exists, please create one).
- [x] Any newly added `unsafe` code is properly documented.

### CODE REVIEW

**Comment by TimePrinciple** [comment]
This search result is from v6.11 kernel:

截屏2024-09-26 13 11 49

File: src/ioctls/vcpu.rs [hunk 1]
```diff
@@ -220,10 +220,9 @@ impl VcpuFd {
     /// let kvm = Kvm::new().unwrap();
     /// let vm = kvm.create_vm().unwrap();
     /// let vcpu = vm.create_vcpu(0).unwrap();
-    /// #[cfg(not(any(target_arch = "arm", target_arch = "aarch64")))]
     /// let regs = vcpu.get_regs().unwrap();
     /// ```
-    #[cfg(not(any(target_arch = "arm", target_arch = "aarch64")))]
+    #[cfg(not(any(target_arch = "arm", target_arch = "aarch64", target_arch = "riscv64")))]
```
Comment (roypat) [inline src/ioctls/vcpu.rs]: Let's hold off on adding the riscv64 directives until #275

Comment (TimePrinciple) [reply src/ioctls/vcpu.rs]: Noted 👍

**Review by roypat — APPROVED** [review]
Thanks for fixing the pit2 thingy!

**Review by ShadowCurse — APPROVED** [review]

**Pull request merged**